internal/dataplane/dataplane_tpl.go

```diff
@@ -0,0 +1,226 @@
+	"consul": {{ .ConsulJSON }},
+	"service": {{ .ServiceJSON }},
+	"logging": {{ .LoggingJSON }},
+	"xdsServer": {{ .XDSServerJSON }},
+	"envoy": {{ .EnvoyJSON }},
+	"telemetry": {{ .TelemetryJSON }}
+}`
+
+type consulConfig struct {
+	addresses       string
+	grpcPort        int
+	skipServerWatch bool
+	tls             *tlsConfig
+	credentials     *credentialsConfig
+}
+
+type consulConfigTmplArgs struct {
+	Addresses       string
+	GRPCPort        int
+	SkipServerWatch bool
+	TLSJSON         string
+	CredentialsJSON string
+}
+
+func (c consulConfig) generateJSON() (string, error) {
+	tmplArgs := &consulConfigTmplArgs{
+		Addresses:       c.addresses,
+		GRPCPort:        c.grpcPort,
+		SkipServerWatch: c.skipServerWatch,
+	}
+
+	if c.tls != nil {
+		tmplArgs.TLSJSON = c.tls.generateJSON()
+	}
+
+	if c.credentials != nil {
+		tmplArgs.CredentialsJSON = c.credentials.generateJSON()
+	}
+
+	t, err := template.New("consulConfig").Parse(consulConfigTemplate)
+	if err != nil {
+		return "", err
+	}
+
+	var buf bytes.Buffer
+	err = t.Execute(&buf, tmplArgs)
+	if err != nil {
+		return "", err
+	}
+
+	return buf.String(), err
+}
+
+const consulConfigTemplate = `{
+	"addresses": "{{ .Addresses }}",
+	"grpcPort": {{ .GRPCPort }},
+	"serverWatchDisabled": {{ .SkipServerWatch }}
+	{{- if .TLSJSON }}
+	,
+	"tls": {{ .TLSJSON }}
+	{{- end }}
+	{{- if .CredentialsJSON }}
+	,
+	"credentials": {{ .CredentialsJSON }}
+	{{- end }}
+}`
+
+type tlsConfig struct {
+	grpcCACertPath string
+	tlsServerName  string
+}
+
+func (t *tlsConfig) generateJSON() string {
+	return `{
+		"disabled": false,
+		"caCertsPath": "` + t.grpcCACertPath + `",
+		"tlsServerName": "` + t.tlsServerName + `"
+	}`
+}
+
+type credentialsConfig struct {
+	credentialType string
+	static         staticCredentialConfig
+}
+
+func (c *credentialsConfig) generateJSON() string {
+	return `{
+		"type": "` + c.credentialType + `",
+		"static":` + c.static.generateJSON() + `
+	}`
+}
+
+type staticCredentialConfig struct {
+	token string
+}
+
+func (s staticCredentialConfig) generateJSON() string {
+	return `{
+		"token": "` + s.token + `"
+	}`
+}
+
+type serviceConfig struct {
+	nodeName       string
+	proxyServiceID string
+	namespace      string
+	partition      string
+}
+
+func (s serviceConfig) generateJSON() string {
+	return `{
+		"nodeName": "` + s.nodeName + `",
+		"serviceID": "` + s.proxyServiceID + `",
+		"namespace": "` + s.namespace + `",
+		"partition": "` + s.partition + `"
+	}`
+}
+
+type loggingConfig struct {
+	json  bool
+	level string
+}
+
+func (l loggingConfig) generateJSON() string {
+	return `{
+		"logLevel": "` + l.level + `",
+		"logJSON": ` + strconv.FormatBool(l.json) + `
+	}`
+}
+
+type xdsServerConfig struct {
+	address string
+	port    int
+}
+
+func (x xdsServerConfig) generateJSON() string {
+	return `{
+		"bindAddress": "` + x.address + `",
+		"bindPort": ` + strconv.FormatInt(int64(x.port), 10) + `
+	}`
+}
+
+type envoyConfig struct {
```

Comment:
We can probably remove this `envoyConfig` struct, because consul-dataplane defaults to `127.0.0.1:19000`.

Users can change this with environment variables, if needed (`DP_ENVOY_ADMIN_BIND_ADDRESS`, `DP_ENVOY_ADMIN_BIND_PORT`).